src/main/java/ca/uhn/fhir/jpa/starter/CollectDataResourceProvider.java

```diff
@@ -0,0 +1,155 @@
+
+        for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
+            if (entry.getResource() instanceof Observation) {
+                Observation resource = (Observation) entry.getResource();
+                Reference subjectRef = resource.getSubject();
+                Reference encounterRef = resource.getEncounter();
+                references.add(subjectRef);
+                references.add(encounterRef);
+            }
+        }
+        report.setEvaluatedResource(references);
+        return report;
+    }
+
+    public Bundle createObsBundle(String url) throws ClientProtocolException, IOException {
+        Bundle observationBundle = new Bundle();
+        Bundle bundle = fetchBundle(url);
+        observationBundle.setType(Bundle.BundleType.COLLECTION);
+
+        for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
+            if (entry.getResource() instanceof Observation) {
+                Resource resource = entry.getResource();
+                Bundle.BundleEntryComponent component = observationBundle.addEntry();
+                component.setResource(resource);
+            }
+        }
+        return observationBundle;
+    }
+
+    private String constructUrl(HttpServletRequest theServletRequest, Measure measure, String periodStart,
+            String periodEnd) {
+        StringBuffer baseUrl = new StringBuffer(theServletRequest.getScheme());
+        baseUrl.append("://");
+        baseUrl.append(theServletRequest.getServerName());
+        baseUrl.append(":");
+        baseUrl.append(theServletRequest.getServerPort());
+
+        String expression = measure.getGroupFirstRep().getPopulationFirstRep().getCriteria().getExpression();
+
+        StringBuffer params = new StringBuffer("&date=ge");
+        params.append(periodStart);
+        params.append("&date=lt");
+        params.append(periodEnd);
+
+        String fullUrl = baseUrl + "/fhir/" + expression + params;
+
+        return fullUrl;
+    }
+
+    private Bundle fetchBundle(String theUrl) throws IOException, ClientProtocolException {
+        Bundle bundle;
+        ourHttpClient = HttpClientBuilder.create().build();
+        HttpGet get = new HttpGet(theUrl);
+
+        String username = "hapi";
+        String password = "hapi123";
+
+        String auth = username + ":" + password;
+        byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+        String authHeader = "Basic " + new String(encodedAuth);
+        get.addHeader(HttpHeaders.AUTHORIZATION, authHeader);
+
+        get.addHeader(Constants.HEADER_CACHE_CONTROL, Constants.CACHE_CONTROL_NO_CACHE);
+        CloseableHttpResponse resp = ourHttpClient.execute(get);
+
+        try {
+            bundle = EncodingEnum.JSON.newParser(ctx).parseResource(Bundle.class,
+                    IOUtils.toString(resp.getEntity().getContent(), Charsets.UTF_8));
+        } finally {
+            IOUtils.closeQuietly(resp);
+        }
```

Comment:
This doesn't rely on the deprecated `IOUtils.closeQuietly()` method and uses `EntityUtils.toString()`, which should properly account for the server-returned content encoding (in case the server explicitly doesn't return UTF-8).

```suggestion
        try (CloseableHttpResponse resp = ourHttpClient.execute(get)) {
            bundle = CONTEXT.newJsonParser().parseResource(Bundle.class,
                    EntityUtils.toString(resp.getEntity(), Charsets.UTF_8));
        }
```